File: C# Examples/12 Working with File References/ListReferences/SolidWorksMacro.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows;
using System.Windows.Forms;

using SolidWorks.Interop.sldworks;
using SolidWorks.Interop.swconst;
using System.Data;


namespace ListReferences
{
    public partial class SolidWorksMacro
    {
        public void Main()
        {
            References.m_swApp = swApp;
            using (Dialog1 dialog = new Dialog1())
            {
                DataTable refs = References.ReadRefs();
                dialog.dataGridView1.DataSource = refs;
                dialog.ShowDialog();
            }

            return;
        }

        // The SldWorks swApp variable is pre-assigned for you.
        public SldWorks swApp;

    }
}

File: 12 Working with File References/ListReferences/SolidWorksMacro.vb
Imports SolidWorks.Interop.sldworks
Imports SolidWorks.Interop.swconst
Imports System.Runtime.InteropServices
Imports System
Imports System.Data

Partial Class SolidWorksMacro

  Public Sub main()

    Dim refDataTable As DataTable
    ReferencesMod.m_swApp = swApp
    refDataTable = ReferencesMod.ReadRefs()
    Dim refDia As New Dialog1
    refDia.FilesGridView.DataSource = refDataTable
    refDia.ShowDialog()

  End Sub

  ''' <summary>
  ''' The SldWorks swApp variable is pre-assigned for you.
  ''' </summary>
  Public swApp As SldWorks


End Class
